packages/state-transition/src/cache/epochCache.ts

```diff
@@ -338,16 +376,40 @@ export class EpochCache {
       throw Error("totalActiveBalanceIncrements >= Number.MAX_SAFE_INTEGER. MAX_EFFECTIVE_BALANCE is too low.");
     }
 
-    const currentShuffling =
-      cachedCurrentShuffling ??
-      computeEpochShuffling(state, currentActiveIndices, currentActiveIndices.length, currentEpoch);
-    const previousShuffling =
-      cachedPreviousShuffling ??
-      (isGenesis
-        ? currentShuffling
-        : computeEpochShuffling(state, previousActiveIndices, previousActiveIndices.length, previousEpoch));
-    const nextShuffling =
-      cachedNextShuffling ?? computeEpochShuffling(state, nextActiveIndices, nextActiveIndices.length, nextEpoch);
+    let previousShuffling: EpochShuffling;
+    let currentShuffling: EpochShuffling;
+    let nextShuffling: EpochShuffling;
+    if (!shufflingCache) {
+      // Only for testing. shufflingCache should always be available in prod
+      previousShuffling = computeEpochShuffling(state, previousActiveIndices, previousEpoch);
+      currentShuffling = isGenesis
+        ? previousShuffling
+        : computeEpochShuffling(state, currentActiveIndices, currentEpoch);
+      nextShuffling = computeEpochShuffling(state, nextActiveIndices, nextEpoch);
+    } else {
+      currentShuffling = cachedCurrentShuffling
+        ? cachedCurrentShuffling
+        : shufflingCache.getSync(currentEpoch, currentDecisionRoot, {
+            state,
+            activeIndices: currentActiveIndices,
+          });
+
+      previousShuffling = cachedPreviousShuffling
+        ? cachedPreviousShuffling
+        : isGenesis
+          ? currentShuffling // TODO: (@matthewkeil) does this need to be added to the cache at previousEpoch and previousDecisionRoot?
```

Comment:
shuffling is mostly used for duties so it's no use to add to the cache
no validator will ask for duties at epoch `-1`

Reply:
Ok, makes sense